rpm/0091-sailfishos-gecko-Initialise-SVGGeometryProperty-Reso.patch

```diff
@@ -0,0 +1,246 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: David Llewellyn-Jones <david.llewellyn-jones@jolla.com>
+Date: Wed, 9 Mar 2022 22:16:56 +0000
+Subject: [PATCH] [sailfishos][gecko] Initialise
+ SVGGeometryProperty::ResolveAll parameters. JB#56892
+
+During SVG rendering HasValidDimensions() was sporadically returning
+false, causing some SVG elements not to render correctly.
+
+The problem appears to be an optimisation glitch. This change
+initialises the parameters passed by reference to
+SVGGeometryProperty::ResolveAll, so that the compiler doesn't determine
+that they may be undefined.
+---
+ dom/svg/SVGCircleElement.cpp           | 4 +++-
+ dom/svg/SVGEllipseElement.cpp          | 3 +++
+ dom/svg/SVGForeignObjectElement.cpp    | 2 ++
+ dom/svg/SVGImageElement.cpp            | 2 ++
+ dom/svg/SVGRectElement.cpp             | 5 +++++
+ layout/svg/nsSVGForeignObjectFrame.cpp | 4 ++++
+ layout/svg/nsSVGImageFrame.cpp         | 6 ++++++
+ 7 files changed, 25 insertions(+), 1 deletion(-)
+
+diff --git a/dom/svg/SVGCircleElement.cpp b/dom/svg/SVGCircleElement.cpp
+index 6fc8d29c2e7c..bba2a93b9a65 100644
+--- a/dom/svg/SVGCircleElement.cpp
++++ b/dom/svg/SVGCircleElement.cpp
+@@ -70,7 +70,7 @@ already_AddRefed<DOMSVGAnimatedLength> SVGCircleElement::R() {
+ 
+ /* virtual */
+ bool SVGCircleElement::HasValidDimensions() const {
+-  float r;
++  float r = 1;
```

Comment:
Why r = 1; instead of r = 0.0f;

Reply:
It's so that, in the event that `r` isn't touched by the call to `ResolveAll`, the condition that follows defaults to `true` and the object is rendered by default:
```
  return r > 0;
```
Since I didn't dig deep enough to know whether the problem is due to the value being left uninitialised, or an incorrect optimisation due to the compiler only *thinking* that it's uninitialised, I thought it better to err on the side of caution and cover both possibilities.

You're right it should at least be `r = 1.0f;` though; I'll fix this.